### ISSUE TITLE
error: call of overloaded 'ResizeBilinearOpModel(<brace-enclosed initializer list>)' is ambiguous

### ISSUE THREAD

**Issue by herbakamil** [tensorflow/tensorflow#26315]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
-  OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 16.04.5 LTS - docker images tensorflow/tensorflow:latest and tensorflow/devel
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: -
- TensorFlow installed from (source or binary): source
- TensorFlow version: commit a2bb5db1bf7931b0dc2cd08e53b8798489568198
- Python version: Python 2.7.12
- Installed using virtualenv? pip? conda?: sources
- Bazel version (if compiling from source): 0.19.2
- GCC/Compiler version (if compiling from source): gcc (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
- CUDA/cuDNN version: -
- GPU model and memory: -

**Describe the problem**
When building tensorflow/lite/kernels:resize_bilinear_test error occurs because there's ambiguity in constructor arguments and as the result which constructor to be used.
```
tensorflow/lite/kernels/resize_bilinear_test.cc:403:58: error: call of overloaded 'ResizeBilinearOpModel(<brace-enclosed initializer list>)' is ambiguous
   ResizeBilinearOpModel m({TensorType_INT8, {1, 2, 2, 2}});
                                                          ^
tensorflow/lite/kernels/resize_bilinear_test.cc:29:12: note: candidate: tflite::{anonymous}::ResizeBilinearOpModel::ResizeBilinearOpModel(const tflite::TensorData&, std::initializer_list<int>)
   explicit ResizeBilinearOpModel(const TensorData& input,
            ^
tensorflow/lite/kernels/resize_bilinear_test.cc:27:7: note: candidate: tflite::{anonymous}::ResizeBilinearOpModel::ResizeBilinearOpModel(const tflite::{anonymous}::ResizeBilinearOpModel&) <deleted>
 class ResizeBilinearOpModel : public SingleOpModel {
       ^
tensorflow/lite/kernels/resize_bilinear_test.cc:27:7: note: candidate: tflite::{anonymous}::ResizeBilinearOpModel::ResizeBilinearOpModel(tflite::{anonymous}::ResizeBilinearOpModel&&) <deleted>
Target //tensorflow/lite/kernels:resize_bilinear_test failed to build
```
It's explained on [stack overflow](https://stackoverflow.com/questions/48011854/deleting-overloaded-function-c11-call-of-overloaded-is-ambiguous/48012069#48012069) and [second point of the document](https://timsong-cpp.github.io/cppwp/n3337/dcl.fct.def.delete)

It can be solved by changing places like:
```
 ResizeBilinearOpModel m({TensorType_FLOAT32, {1, 2, 1, 1}});
```
to 
```
 ResizeBilinearOpModel m(TensorData({TensorType_FLOAT32, {1, 2, 1, 1}}));
```

**Provide the exact sequence of commands / steps that you executed before running into the problem**
bazel test --config=opt tensorflow/lite/kernels:resize_bilinear_test

**Any other info / logs**
Below is a simplified code showing the same problem as occurring inside the test:
```
struct Pair{
	int i;
	int j;
	Pair(int i, int j = -1): i(i), j(j) {};
};

struct B {
	B(const B&) = delete;
	explicit B(const Pair& p, int j = 0) {
		cout << p.i << " " << p.j << " " << j << endl;
	}
};

int main() {
	B b({1, 2});
}
```
It also causes:
```
delete.cpp: In function ‘int main()’:
delete.cpp:32:12: error: call of overloaded ‘B(<brace-enclosed initializer list>)’ is ambiguous
  B b({1, 2});
            ^
delete.cpp:20:11: note: candidate: B::B(const Pair&, int)
  explicit B(const Pair& p, int j = 0) {
           ^
delete.cpp:19:2: note: candidate: B::B(const B&) <deleted>
  B(const B&) = delete;
  ^
```
It doesn't matter that copy constructor is deleted as this resolution takes place later.

**Comment by herbakamil** [comment]
@shashishekhar you've recently worked on this code. Is it tested on some internal Google compiler which behaves differently form gcc on official TF docker image?
If yes are there any plans to apply in the future compiler compatible with gcc?

**Comment by mohantym** [comment]
Hi @herbakamil !
This issue has been addressed in the 2.9 version. Attached [gist](https://colab.sandbox.google.com/gist/mohantym/b635e84d3cbf499a36828a2abea504d4/tensorflow-ranking.ipynb#scrollTo=skRg4_WPi616) for reference.
Thank you!
